### DISCUSSION TITLE
Linux watchdog support

### DISCUSSION THREAD

**Discussion by mbrainin** [polarfire-soc/polarfire-soc-documentation#351] [HSS and U-Boot]
Hi,

According to https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/hss-and-u-boot/watchdog-service.md, support for the Linux watchdog kernel driver is still under development. Is that still the case or is the driver supported now?
Also, if it is not supported, does that mean the HSS can setup the watchdog to monitor all the cores, but the cores that Linux is running on will not be able to update the refresh the watchdog timer resulting in the HSS rebooting?

**Comment by ConchuOD** [comment]
Hey,

There is not yet a Linux driver for the watchdog, despite that comment
in the documentation. Our upcoming release is planned for September,
but we've already got our list of changes in mind so we'll add the
watchdog driver to the todo list for the release after that.

In the meantime, you can use something like the devmem tool to control
the watchdog from Linux userspace. If you're doing so, make sure to
read the spec carefully, so as to avoid any unintended triggering!

> Also, if it is not supported, does that mean the HSS can setup the
> watchdog to monitor all the cores, but the cores that Linux is running
> on will not be able to update the refresh the watchdog timer resulting
> in the HSS rebooting?

Unless the Linux context goes and enables the watchdog, there's nothing
to fear. If the watchdog is enabled from userspace, then userspace would
have to keep tabs on it so that it does not trigger, in which case, yes
the HSS would trigger a reboot. Whether that reboot is of just the context
or the whole system depends on the configuration of the HSS.

Hope that helps,
Conor.
